cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -384,7 +385,20 @@ private boolean isDeleteLifecycleEnabled(String bucketName) {
       }
       if (rule.getAction() instanceof BucketInfo.LifecycleRule.DeleteLifecycleAction &&
         rule.getCondition().getDaysSinceCustomTime() > 0) {
-        return true;
+        try {
+          int artifactsComputeHashTimeBucketDays = Integer.parseInt(
+            provisionerContext.getProperties().get(DataprocUtils.ARTIFACTS_COMPUTE_HASH_TIME_BUCKET_DAYS));
```

Comment:
Check for cases in which ARTIFACTS_COMPUTE_HASH_TIME_BUCKET_DAYS does not exist

Reply:
`timeBucketDays` instead of artifactsComputeHashTimeBucketDays?

Reply:
Here you are assuming that whatever call dataproc manager has this key/value set in properties, but there is no contract (or language construct) enforcing it. So if it mistakenly gets deleted in DistributedProgramRunner, it will be caught during runtime.

Reply:
Added a `contains` check.